### ISSUE TITLE
Plugins CI is failing to build the example APKs

### ISSUE THREAD

**Issue by mklim** [flutter/flutter#26925]
There's a problem in Cirrus with dependencies on conflicting versions of `com.android.support`.

```
Execution failed for task ':app:lintVitalRelease'.
> Could not resolve all artifacts for configuration ':app:debugAndroidTestRuntimeClasspath'.
   > Could not resolve com.android.support:support-annotations:26.1.0.
     Required by:
         project :app
      > Cannot find a version of 'com.android.support:support-annotations' that satisfies the version constraints: 
           Dependency path 'android:app:unspecified' --> 'com.android.support.test:runner:1.0.2' --> 'com.android.support:support-annotations:27.1.1'
           Dependency path 'android:app:unspecified' --> 'com.android.support.test:runner:1.0.2' --> 'com.android.support.test:monitor:1.0.2' --> 'com.android.support:support-annotations:27.1.1'
           Dependency path 'android:app:unspecified' --> 'com.android.support.test.espresso:espresso-core:3.0.2' --> 'com.android.support.test:runner:1.0.2' --> 'com.android.support:support-annotations:27.1.1'
           Dependency path 'android:app:unspecified' --> 'io.flutter.plugins.firebaseauth:firebase_auth:1.0-SNAPSHOT' --> 'com.google.firebase:firebase-auth:16.0.4' --> 'com.google.android.gms:play-services-basement:16.0.1' --> 'com.android.support:support-v4:26.1.0' --> 'com.android.support:support-compat:26.1.0' --> 'com.android.support:support-annotations:26.1.0'
           Dependency path 'android:app:unspecified' --> 'io.flutter.plugins.firebaseauth:firebase_auth:1.0-SNAPSHOT' --> 'com.google.firebase:firebase-auth:16.0.4' --> 'com.google.android.gms:play-services-basement:16.0.1' --> 'com.android.support:support-v4:26.1.0' --> 'com.android.support:support-media-compat:26.1.0' --> 'com.android.support:support-annotations:26.1.0'
           Dependency path 'android:app:unspecified' --> 'io.flutter.plugins.firebaseauth:firebase_auth:1.0-SNAPSHOT' --> 'com.google.firebase:firebase-auth:16.0.4' --> 'com.google.android.gms:play-services-basement:16.0.1' --> 'com.android.support:support-v4:26.1.0' --> 'com.android.support:support-core-utils:26.1.0' --> 'com.android.support:support-annotations:26.1.0'
           Dependency path 'android:app:unspecified' --> 'io.flutter.plugins.firebaseauth:firebase_auth:1.0-SNAPSHOT' --> 'com.google.firebase:firebase-auth:16.0.4' --> 'com.google.android.gms:play-services-basement:16.0.1' --> 'com.android.support:support-v4:26.1.0' --> 'com.android.support:support-core-ui:26.1.0' --> 'com.android.support:support-annotations:26.1.0'
           Constraint path 'android:app:unspecified' --> 'com.android.support:support-annotations' strictly '26.1.0' because of the following reason: debugRuntimeClasspath uses version 26.1.0
           Constraint path 'android:app:unspecified' --> 'com.android.support:support-annotations' strictly '26.1.0' because of the following reason: debugRuntimeClasspath uses version 26.1.0
           Constraint path 'android:app:unspecified' --> 'com.android.support:support-annotations' strictly '26.1.0' because of the following reason: debugRuntimeClasspath uses version 26.1.0
           Constraint path 'android:app:unspecified' --> 'com.android.support:support-annotations' strictly '26.1.0' because of the following reason: debugRuntimeClasspath uses version 26.1.0
           Constraint path 'android:app:unspecified' --> 'com.android.support:support-annotations' strictly '26.1.0' because of the following reason: debugRuntimeClasspath uses version 26.1.0
           Constraint path 'android:app:unspecified' --> 'com.android.support:support-annotations' strictly '26.1.0' because of the following reason: debugRuntimeClasspath uses version 26.1.0
   > Could not resolve com.android.support:support-annotations:27.1.1.
     Required by:
         project :app > com.android.support.test:runner:1.0.2
         project :app > com.android.support.test:runner:1.0.2 > com.android.support.test:monitor:1.0.2
      > Cannot find a version of 'com.android.support:support-annotations' that satisfies the version constraints: 
           Dependency path 'android:app:unspecified' --> 'com.android.support.test:runner:1.0.2' --> 'com.android.support:support-annotations:27.1.1'
           Dependency path 'android:app:unspecified' --> 'com.android.support.test:runner:1.0.2' --> 'com.android.support.test:monitor:1.0.2' --> 'com.android.support:support-annotations:27.1.1'
           Dependency path 'android:app:unspecified' --> 'com.android.support.test.espresso:espresso-core:3.0.2' --> 'com.android.support.test:runner:1.0.2' --> 'com.android.support:support-annotations:27.1.1'
           Dependency path 'android:app:unspecified' --> 'io.flutter.plugins.firebaseauth:firebase_auth:1.0-SNAPSHOT' --> 'com.google.firebase:firebase-auth:16.0.4' --> 'com.google.android.gms:play-services-basement:16.0.1' --> 'com.android.support:support-v4:26.1.0' --> 'com.android.support:support-compat:26.1.0' --> 'com.android.support:support-annotations:26.1.0'
           Dependency path 'android:app:unspecified' --> 'io.flutter.plugins.firebaseauth:firebase_auth:1.0-SNAPSHOT' --> 'com.google.firebase:firebase-auth:16.0.4' --> 'com.google.android.gms:play-services-basement:16.0.1' --> 'com.android.support:support-v4:26.1.0' --> 'com.android.support:support-media-compat:26.1.0' --> 'com.android.support:support-annotations:26.1.0'
           Dependency path 'android:app:unspecified' --> 'io.flutter.plugins.firebaseauth:firebase_auth:1.0-SNAPSHOT' --> 'com.google.firebase:firebase-auth:16.0.4' --> 'com.google.android.gms:play-services-basement:16.0.1' --> 'com.android.support:support-v4:26.1.0' --> 'com.android.support:support-core-utils:26.1.0' --> 'com.android.support:support-annotations:26.1.0'
           Dependency path 'android:app:unspecified' --> 'io.flutter.plugins.firebaseauth:firebase_auth:1.0-SNAPSHOT' --> 'com.google.firebase:firebase-auth:16.0.4' --> 'com.google.android.gms:play-services-basement:16.0.1' --> 'com.android.support:support-v4:26.1.0' --> 'com.android.support:support-core-ui:26.1.0' --> 'com.android.support:support-annotations:26.1.0'
           Constraint path 'android:app:unspecified' --> 'com.android.support:support-annotations' strictly '26.1.0' because of the following reason: debugRuntimeClasspath uses version 26.1.0
           Constraint path 'android:app:unspecified' --> 'com.android.support:support-annotations' strictly '26.1.0' because of the following reason: debugRuntimeClasspath uses version 26.1.0
           Constraint path 'android:app:unspecified' --> 'com.android.support:support-annotations' strictly '26.1.0' because of the following reason: debugRuntimeClasspath uses version 26.1.0
           Constraint path 'android:app:unspecified' --> 'com.android.support:support-annotations' strictly '26.1.0' because of the following reason: debugRuntimeClasspath uses version 26.1.0
           Constraint path 'android:app:unspecified' --> 'com.android.support:support-annotations' strictly '26.1.0' because of the following reason: debugRuntimeClasspath uses version 26.1.0
           Constraint path 'android:app:unspecified' --> 'com.android.support:support-annotations' strictly '26.1.0' because of the following reason: debugRuntimeClasspath uses version 26.1.0
   > Could not resolve com.android.support:support-annotations:26.1.0.
     Required by:
         project :app > com.android.support:support-compat:26.1.0
         project :app > com.android.support:support-media-compat:26.1.0
         project :app > com.android.support:support-core-utils:26.1.0
         project :app > com.android.support:support-core-ui:26.1.0
      > Cannot find a version of 'com.android.support:support-annotations' that satisfies the version constraints: 
           Dependency path 'android:app:unspecified' --> 'com.android.support.test:runner:1.0.2' --> 'com.android.support:support-annotations:27.1.1'
           Dependency path 'android:app:unspecified' --> 'com.android.support.test:runner:1.0.2' --> 'com.android.support.test:monitor:1.0.2' --> 'com.android.support:support-annotations:27.1.1'
           Dependency path 'android:app:unspecified' --> 'com.android.support.test.espresso:espresso-core:3.0.2' --> 'com.android.support.test:runner:1.0.2' --> 'com.android.support:support-annotations:27.1.1'
           Dependency path 'android:app:unspecified' --> 'io.flutter.plugins.firebaseauth:firebase_auth:1.0-SNAPSHOT' --> 'com.google.firebase:firebase-auth:16.0.4' --> 'com.google.android.gms:play-services-basement:16.0.1' --> 'com.android.support:support-v4:26.1.0' --> 'com.android.support:support-compat:26.1.0' --> 'com.android.support:support-annotations:26.1.0'
           Dependency path 'android:app:unspecified' --> 'io.flutter.plugins.firebaseauth:firebase_auth:1.0-SNAPSHOT' --> 'com.google.firebase:firebase-auth:16.0.4' --> 'com.google.android.gms:play-services-basement:16.0.1' --> 'com.android.support:support-v4:26.1.0' --> 'com.android.support:support-media-compat:26.1.0' --> 'com.android.support:support-annotations:26.1.0'
           Dependency path 'android:app:unspecified' --> 'io.flutter.plugins.firebaseauth:firebase_auth:1.0-SNAPSHOT' --> 'com.google.firebase:firebase-auth:16.0.4' --> 'com.google.android.gms:play-services-basement:16.0.1' --> 'com.android.support:support-v4:26.1.0' --> 'com.android.support:support-core-utils:26.1.0' --> 'com.android.support:support-annotations:26.1.0'
           Dependency path 'android:app:unspecified' --> 'io.flutter.plugins.firebaseauth:firebase_auth:1.0-SNAPSHOT' --> 'com.google.firebase:firebase-auth:16.0.4' --> 'com.google.android.gms:play-services-basement:16.0.1' --> 'com.android.support:support-v4:26.1.0' --> 'com.android.support:support-core-ui:26.1.0' --> 'com.android.support:support-annotations:26.1.0'
           Constraint path 'android:app:unspecified' --> 'com.android.support:support-annotations' strictly '26.1.0' because of the following reason: debugRuntimeClasspath uses version 26.1.0
           Constraint path 'android:app:unspecified' --> 'com.android.support:support-annotations' strictly '26.1.0' because of the following reason: debugRuntimeClasspath uses version 26.1.0
           Constraint path 'android:app:unspecified' --> 'com.android.support:support-annotations' strictly '26.1.0' because of the following reason: debugRuntimeClasspath uses version 26.1.0
           Constraint path 'android:app:unspecified' --> 'com.android.support:support-annotations' strictly '26.1.0' because of the following reason: debugRuntimeClasspath uses version 26.1.0
           Constraint path 'android:app:unspecified' --> 'com.android.support:support-annotations' strictly '26.1.0' because of the following reason: debugRuntimeClasspath uses version 26.1.0
           Constraint path 'android:app:unspecified' --> 'com.android.support:support-annotations' strictly '26.1.0' because of the following reason: debugRuntimeClasspath uses version 26.1.0
* Try:
Run with --stacktrace option to get the stack trace. Run with --info or --debug option to get more log output. Run with --scan to get full insights.
* Get more help at https://help.gradle.org
BUILD FAILED in 33s
```

So far I've been unable to reproduce the actual crash locally, but I can consistently see the version that's being successfully required and found on my own machine by moving into `example/android/app` in any given package and then running `/.gradlew --debug :app:lintVitalRelease`. So far I'm only seeing `26` depended on locally.

Fixing #23995 is generally desirable and would have the added benefit of avoiding this problem entirely. However this would be both a risky migration and a potentially breaking change for all of the plugins users that are currently relying on the support libraries. Moving forward with it because of a blocking problem in our CI seems ill advised.

flutter/plugins#1091 is a WIP PR trying to address this.

/cc @dnfield @amirh

**Comment by mklim** [comment]
I threw together a quick patch to migrate the plugins off of the android support libraries. flutter/plugins#1103.

The build APK step works locally but is failing with a new error in CI:

```
FAILURE: Build failed with an exception.
* What went wrong:
A problem occurred configuring project ':video_player'.
> Failed to install the following Android SDK packages as some licences have not been accepted.
     platforms;android-27 Android SDK Platform 27
  To build this project, accept the SDK license agreements and install the missing components using the Android Studio SDK Manager.
  Alternatively, to transfer the license agreements from one workstation to another, see http://d.android.com/r/studio-ui/export-licenses.html
  Using Android SDK: /opt/android-sdk-linux
* Try:
Run with --stacktrace option to get the stack trace. Run with --info or --debug option to get more log output. Run with --scan to get full insights.
* Get more help at https://help.gradle.org
BUILD FAILED in 7s
```

**Comment by mklim** [comment]
Android issues have been fixed by flutter/plugins#1104
